src/theme/components/hero.ts

```diff
@@ -105,8 +105,65 @@ const getSecondaryVariantStyles = (bgColor: string = "") => {
     },
   };
 };
+const getTextColor = (type, mode, foregroundColor, isDarkText) => {
+  const prefix = mode === "dark" ? "dark.ui." : "ui.";
+  const colorLight = foregroundColor
+    ? foregroundColor
+    : isDarkText
+    ? `${prefix}typography.${type}`
+    : `${prefix}typography.inverse.${type}`;
+  const colorDark = foregroundColor
+    ? foregroundColor
+    : isDarkText
+    ? `${prefix}typography.inverse.${type}`
+    : `${prefix}typography.${type}`;
+  const colorLink = foregroundColor
+    ? foregroundColor
+    : isDarkText
+    ? `ui.link.primary`
+    : `dark.ui.link.primary`;
+  const finalColor =
+    type === "link" ? colorLink : mode === "dark" ? colorDark : colorLight;
+  return finalColor;
+};
+const getLinkColor = (state, mode, foregroundColor, isDarkText) => {
+  let finalColor;
+  const isDarkMode = mode === "dark" ? true : false;
+  switch (state) {
+    default:
+      finalColor = foregroundColor
+        ? foregroundColor
+        : isDarkText
+        ? `var(--nypl-colors-ui-link-primary) !important`
+        : isDarkMode
+        ? `dark.ui.link.primary`
+        : `dark.ui.link.primary`;
+      break;
+    case "hover": {
+      finalColor = foregroundColor
+        ? foregroundColor
+        : isDarkText
+        ? `var(--nypl-colors-ui-link-secondary) !important`
+        : isDarkMode
+        ? `dark.ui.link.secondary`
+        : `dark.ui.link.secondary`;
+      break;
+    }
+    case "visited": {
+      finalColor = foregroundColor
+        ? foregroundColor
+        : isDarkText
+        ? `var(--nypl-colors-ui-link-tertiary) !important`
+        : isDarkMode
+        ? `dark.ui.link.tertiary`
```

Comment:
We don't have any rules or patterns for this but I'd like to keep tertiary to two levels or else it gets confusing to read. That said, I'm sure we do this in other files so it's not something you need to change but just to keep in mind for readability.

Reply:
When I put this together, I felt like I was being bad.  But it worked, so I left it.  That said, I simplified the statement and removed the last ternary condition.